es/separable_natural_es.py

```diff
@@ -24,21 +27,32 @@ def optimize(func, mu, sigma,
     if population_size is None:
         population_size = lib.default_population_size(mu.size)
 
+    if rng is None:
+        rng = np.random.RandomState()
+    elif isinstance(rng, int):
+        rng = np.random.RandomState(seed=rng)
+
     generation = 0
     history_mu = []
     history_sigma = []
     history_pop = []
+    history_fitness = []
 
     while True:
-        s = np.random.normal(0, 1, size=(population_size, *np.shape(mu)))
+        s = rng.normal(0, 1, size=(population_size, *np.shape(mu)))
         z = mu + sigma * s
 
         if mirrored_sampling:
             z = np.vstack([z, mu - sigma * s])
             s = np.vstack([s, -s])
-
-        fitness = np.fromiter((func(zi) for zi in z), np.float)
-
+        if parallel_threads is None:
+            fitness = np.fromiter((func(zi) for zi in z), np.float)
+        else:
+            func_map = partial(func)
```

Comment:
why do you need the partial here? you do not pass any arguments so `func_map` has the same signature as `func` doesn't it? or is there some magic going on that should be commented?